shotgun_api3/shotgun.py

```diff
@@ -675,6 +688,32 @@ def _construct_read_parameters(self,
         params["paging"] = { "entities_per_page": self.config.records_per_page,
                              "current_page": 1 }
 
+        if additional_server_presets is not None:
+            if not isinstance(additional_server_presets, list):
+                msg = "'additional_server_presets' parameter must be a list or None"
+                raise ValueError(msg)
+
+            preset_list = []
+            for preset in additional_server_presets:
+                if not isinstance(preset, list):
+                    msg = "Elements of 'additional_server_presets' parameter must be lists"
+                    raise ValueError(msg)
+
+                if len(preset) == 0:
+                    msg = "Missing 'name' for 'additional_server_presets'"
+                    raise ValueError(msg)
+
+                preset_param = {
```

Comment:
nit: Maybe this variable could be named `crud_preset` or something? It's a little confusing that it's named preset_param but then internally contains a 'param's dictionary with the preset params :)